kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchBankInputFileType.java

```diff
@@ -0,0 +1,371 @@
+			bank.setBankOfficeCode(KFSConstants.ACHBankOfficeCodes.MAIN);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INVALID_OFFICE_CD_ERROR));
+		}
+		
+		if (StringUtils.isBlank(bankNewRoutingNbr)) {
+			bank.setBankNewRoutingNumber(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_NEW_ROUTING_NUMBER);
+		}
+		
+		if (StringUtils.isBlank(bankPhoneAreaCd)) {
+			bank.setBankPhoneAreaCode(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_PHONE_AREA_CODE);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INCOMPLETE_PHONE_ERROR));
+		}
+		
+		if (StringUtils.isBlank(bankPhonePrefixNbr)) {
+			bank.setBankPhonePrefixNumber(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_PHONE_PREFIX_NUMBER);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INCOMPLETE_PHONE_ERROR));
+		}
+		
+		if (StringUtils.isBlank(bankPhoneSuffixNbr)) {
+			bank.setBankPhoneSuffixNumber(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_PHONE_SUFFIX_NUMBER);
+			reportLines.add(new BankReportLine(bankRoutingNbr,bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INCOMPLETE_PHONE_ERROR));
+		}
+		
+		if (StringUtils.isBlank(bankInstitutionStatusCd)) {
+			bank.setBankInstitutionStatusCode(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_INSTITUTION_STATUS_CODE);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INVALID_INST_STAT_CD));
+		}
+	
+	
+		if (StringUtils.isBlank(bankDataViewCd)) {
+			bank.setBankDataViewCode(KFSConstants.ACHFileDefaultConstants.DEFAULT_BANK_DATA_VIEW_CODE);
+			reportLines.add(new BankReportLine(bankRoutingNbr, bankNm, KFSConstants.ACHFileConstants.ACH_BANK_INVALID_DATA_VIEW_CD));
+		}
+		
+		return true;
+	}
+		
+	protected void writeReport(List<BankReportLine> lines, String systemSource) {
+		File reportFile = new File(reportPath + "/" + reportPrefix + "_" + dateTimeService.toDateTimeStringForFilename(dateTimeService.getCurrentDate()) + "." + reportExtension);
+		BufferedWriter writer = null;
+		String routingNbr;
+		String bankNm;
+		
+		try {
+			writer = new BufferedWriter(new PrintWriter(reportFile));
+			writeReportTitle(writer, systemSource);
+			writeReportHeadings(writer);
+			for (BankReportLine line : lines) {
+				routingNbr = rightPad(line.getBankRoutingNumber(), 15, " ");
+				bankNm = rightPad(line.getBankName(), 40, " ");
+				writer.write(routingNbr + bankNm + line.getMessage());
+				writer.newLine();
+				writer.newLine();
+			}
+		}
+		catch (FileNotFoundException e) {
+			LOG.error(reportFile + " not found " + " " + e.getMessage());
+			throw new RuntimeException(reportFile + " not found" + e.getMessage(), e);
+		}
+		catch (IOException e) {
+			LOG.error("Error writing to BufferedWriter " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
+		finally {
+			try {
```

Comment:
289-291 could be replaced by one `IOUtils.close(...) call, which would help flow, thus readability.